keep/api/core/db.py

```diff
@@ -2044,6 +2043,42 @@ def get_incident_alerts_by_incident_id(tenant_id: str, incident_id: str) -> List
     return query.all()
 
 
+def get_alerts_data_for_incident(
```

Comment:
Is this performant now? can you add some documentation on what exactly is being done here? why is the inner function needed?